sdk/shared/FreeRTOS_tick_config.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * FreeRTOS Kernel V10.1.1
+ * Copyright (C) 2018 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://www.FreeRTOS.org
+ * http://aws.amazon.com/freertos
+ *
+ * 1 tab == 4 spaces!
+ */
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* Xilinx includes. */
+#include "xscugic.h"
+#include "xscutimer.h"
+
+#define XSCUTIMER_CLOCK_HZ (XPAR_CPU_CORTEXA9_0_CPU_CLK_FREQ_HZ / 2UL)
+
+static XScuTimer xTimer;
+XScuGic xInterruptController; /* Interrupt controller instance */
+
+/*
+ * The application must provide a function that configures a peripheral to
+ * create the FreeRTOS tick interrupt, then define configSETUP_TICK_INTERRUPT()
+ * in FreeRTOSConfig.h to call the function.  This file contains a function
+ * that is suitable for use on the Zynq SoC.
+ */
+void vConfigureTickInterrupt(void)
+{
+    BaseType_t xStatus;
+    extern void FreeRTOS_Tick_Handler(void);
+    XScuTimer_Config *pxTimerConfig;
+    XScuGic_Config *pxGICConfig;
+    const uint8_t ucRisingEdge = 3;
+
+    /* This function is called with the IRQ interrupt disabled, and the IRQ
+    interrupt should be left disabled.  It is enabled automatically when the
+    scheduler is started. */
+
+    /* Ensure XScuGic_CfgInitialize() has been called.  In this demo it has
+    already been called from prvSetupHardware() in main(). */

```

Comment:
Hm I don't see your `prvSetupHardware()` being called in main.c... Does the tick interrupt work for you?

i.e., you can make a task sleep for X seconds and it will wake up correctly? Maybe your blinky does this?

Reply:
Seems like your blinky task is using the FreeRTOS sleep methods, and it works... so probably fine

https://github.com/Severson-Group/AMDC-Firmware/blob/feature/freertos-stable/sdk/freertos_app_cpu0/src/main.c#L284-L298